packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -85,17 +88,20 @@ const BLOCK_EDITOR_SETTINGS = [
 /**
  * React hook used to compute the block editor settings to use for the post editor.
  *
- * @param {Object} settings EditorProvider settings prop.
- * @param {string} postType Editor root level post type.
- * @param {string} postId   Editor root level post ID.
+ * @param {Object} settings        EditorProvider settings prop.
+ * @param {string} postType        Editor root level post type.
+ * @param {string} postId          Editor root level post ID.
+ * @param {string} contextPostType The post type of the edited post.
  *
  * @return {Object} Block Editor Settings.
  */
-function useBlockEditorSettings( settings, postType, postId ) {
+function useBlockEditorSettings( settings, postType, postId, contextPostType ) {
```

Comment:
I think ideally the contextPostType should replace the postType argument, and we could pass a "mode" in addition. That way we can fallback to `wp_template` post type if the mode is "template-locked".